torch/csrc/autograd/python_variable.cpp

```diff
@@ -2079,17 +2099,24 @@ static PyObject* THPVariable_NewWithVar(
         " which is not a subclass of the "
         "requested type");
     // We may (in fact, we typically will) need to resurrect this
-    return THPVariable_Wrap(_var);
+    return THPVariable_Wrap(::std::forward<T>(_var));
   }
 
   PyObject* obj = type->tp_alloc(type, 0);
   if (obj) {
     auto v = (THPVariable*)obj;
-    // TODO: named constructor to avoid default initialization
-    new (&v->cdata) MaybeOwned<Variable>();
-    if (c10::impl::HermeticPyObjectTLS::get_state()) {
+    auto has_state = c10::impl::HermeticPyObjectTLS::get_state();
+    if constexpr (std::is_rvalue_reference_v<decltype(::std::forward<T>(
+                      _var))>) {
+      new (&v->cdata) MaybeOwned<Variable>(
+          MaybeOwned<Variable>::owned(::std::forward<T>(_var)));
+    } else {
+      new (&v->cdata) MaybeOwned<Variable>(
+          MaybeOwned<Variable>::owned(Variable(::std::forward<T>(_var))));
+    }
```

Comment:
beh what happened lol

Reply:
> beh what happened lol

For const TensorBase& , we have to construct a Variable to pass to  MaybeOwned, due to the constructor constraint.